protocol_rfcs/managed-commits.md

```diff
@@ -0,0 +1,315 @@
+```
+
+Let say the commit-owner is tracking:
+```
+{
+  6  -> "00000000000000000006.3a0d65cd-4056-49b8-937b-95f9e3ee90e5.json",
+  7  -> "00000000000000000007.016ae953-37a9-438e-8683-9a9a4a79a395.json",
+  8  -> "00000000000000000008.7d17ac10-5cc3-401b-bd1a-9c82dd2ea032.json",
+  9  -> "00000000000000000009.41bf693a-f5b9-4478-9434-af7475d5a9f0.json"
+}
+```
+
+Delta clients have two choices to read such tables:
+1. A non-managed-commit aware delta implementation can read such table by listing the `_delta_log` directory and reading the delta/checkpoint/log-compaction files.
+   They cannot use the un-backfilled commits in `_delta_log/_commits` directory as they are not aware of the external commit-owner. Because of this, they may see a stale snapshot
+   of the table if the recent commits are not backfilled.
+    - In the above example, such delta implementation will see version 7 as the latest snapshot.
+2. A client that understands managed commits can supplement the file listing result by contacting the commit-owner for information about un-backfilled commits, in order to get the most
+   recent snapshot of the table.
+    - In the above example, a delta implementation could get information about versions 0 through 7 from _delta_log directory and get information about un-backfilled commits (v8, v9) from the commit-owner.
+
+## Maintenance operations on managed-commit tables
+
+Maintenance operations such as [checkpoints](#checkpoints-1), [log compaction](#log-compaction-files), [metadata cleanup](#metadata-cleanup) can be performed only on the backfilled part of a managed-commit table.
+
+- A [checkpoint](#checkpoints-1) for a version `v` can be created only if the commit `v` is already backfilled i.e. `_delta_log/<v>.json` exists.
+- A [log compaction](#log-compaction-files) can be done only on the backfilled part of the table. The log compaction file `<x>.<y>.compacted.json` can be created only if commit `y` is already backfilled i.e. `_delta_log/<y>.json` exists.
+- A [Metadata cleanup](#metadata-cleanup) must always preserve the newest k >= 1 backfilled commits. In other words, [metadata cleanup](#metadata-cleanup) must always choose a backfilled `cutOffCheckpoint`.
+    - This allows the table to remain discoverable by the filesystem-based delta readers even after metadata cleanup deletes everything before commit version `cutOffCheckpoint`.
+
+## Managed Commit Enablement
+
+The managed-commit feature is supported and active when:
+- The table must be on Writer Version 7.
+- The table has a `protocol` action with `writerFeatures` containing the feature `managedCommits`.
+- The table has a metadata property `delta.managedCommitOwnerName` in the [change-metadata](#change-metadata)'s configuration.
+- The table may have a metadata property `delta.managedCommitOwnerConf` in the [change-metadata](#change-metadata)'s configuration. The value of this property is a json-coded string-to-string map.
```

Comment:
Should we do some namespacing here, with e.g. `delta.managedCommits.commitOwner` and `delta.managedCommits.commitOwnerConf`?